### PR TITLE
Update SDK version

### PR DESCRIPTION
**Pull request by michalpelka**
## What does this PR do?

Since some braking changes are introduced to ROS 2 Gem (e.g. #593), we need to update major version of ROS 2 Gem.

## How was this PR tested?

Build against o3de dev branch `496f6130575367a56d99f54038d00818b1800523`

### CODE REVIEW

**Comment by michalpelka** [comment]
According to a discussion in #593 - I will update this PR #593  to the minimum required version of ROS2 gem in project templates.

**Comment by arturkamieniecki** [comment]
I've bumped the ROS2 gem versions and fixed broken tests.

File: Gems/ROS2/gem.json [hunk 1]
```diff
@@ -18,8 +18,8 @@
         "ROS2"
     ],
     "compatible_engines": [
-        "o3de-sdk>=2.1.0",
-        "o3de>=2.1.0"
+        "o3de-sdk>=4.1.0",
+        "o3de>=4.1.0"
```
Comment (spham-amzn) [inline Gems/ROS2/gem.json:22]: Shouldn't this be '3'?

File: Gems/ROS2/gem.json [hunk 2]
```diff
@@ -18,8 +18,8 @@
         "ROS2"
     ],
     "compatible_engines": [
-        "o3de-sdk>=2.1.0",
-        "o3de>=2.1.0"
+        "o3de-sdk>=4.1.0",
```
Comment (michalpelka) [inline Gems/ROS2/gem.json:21]:
```suggestion
        "o3de-sdk>=3.1.0",
        "o3de>=3.1.0"
```

File: Gems/ROS2/gem.json (hunk 1 above)
Comment (michalpelka) [reply Gems/ROS2/gem.json]: Set to 3.1.0. Thanks!

**Review by spham-amzn — APPROVED** [review]

File: Gems/ROS2/gem.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (spham-amzn) [reply Gems/ROS2/gem.json]: I just checked with development, the current engine version is "4.0.0"

Comment (spham-amzn) [reply Gems/ROS2/gem.json]: Sorry, I just checked o3de/development and the engine version is [4.0.0](https://github.com/o3de/o3de/blob/57c811f7fb5505600b081f14bf197b3ce1c26183/engine.json#L7), sorry about that.

**Review by jhanca-robotecai — COMMENTED** [review]
I tested ROS2ProjectTemplate with extra Proteus robot added - works as expected.

I am not sure about the `o3de` version though - based on @spham-amzn check:
> I just checked with development, the current engine version is "4.0.0"

I believe we should target `>=4.0.0`, as the current `development` branch in `extras` will not build correctly with `2310.x` due to linking errors after changes in module names in cmake.

**Review by adamdbrw — APPROVED** [review]

**Pull request merged**